lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -0,0 +1,282 @@
+
+    def test_normalizing_along_another_axis_with_error(self):
+        """Normalizing along another axis, when raises error.
+           In this case we are normalizing along an axis where in some places
+           the sum along that axis is zero."""
+        nbhooded_cube = self.weights_cube.copy()
+        plugin = CollapseMaskedNeighbourhoodCoordinate(
+            "projection_x_coordinate", self.weights_cube)
+        message = "Sum of weights must be > 0.0"
+        with self.assertRaisesRegexp(ValueError, message):
+            plugin.reweight_weights(nbhooded_cube, self.weights_cube)
+
+    def test_normalizing_along_another_axis(self):
+        """Normalizing along another axis, when this is a valid thing to do.
+           This is normalizing along the rows of the input weights."""
+        input_weights = np.array([[[0.0, 0.7, 0., 0, 0],
+                                   [0.0, 0.3, 0, 0, 0],
+                                   [0.3, 0.1, 0, 0, 0],
+                                   [0.2, 0, 0, 0, 0],
+                                   [0.1, 0, 0, 0, 0]],
+                                  [[1.0, 0, 1, 1, 1],
+                                   [1.0, 0, 1, 1, 1],
+                                   [0, 0, 1, 1, 1],
+                                   [0, 0, 0, 0.9, 0.1],
+                                   [1, 1, 1, 0, 0]],
+                                  [[0.2, 0, 0, 0, 0],
+                                   [0.3, 0, 0, 0, 0],
+                                   [0.1, 0, 0, 0, 0],
+                                   [0, 0, 0, 0.1, 0.0],
+                                   [0, 0, 0, 0.4, 0.8]]])
+        expected_weights = np.array([[[0.0, 1, 0., 0, 0],
+                                      [0.0, 1, 0, 0, 0],
+                                      [0.75, 0.25, 0, 0, 0],
+                                      [1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0]],
+                                     [[0.25, 0, 0.25, 0.25, 0.25],
+                                      [0.25, 0, 0.25, 0.25, 0.25],
+                                      [0, 0, 0.333333, 0.333333, 0.333333],
+                                      [0, 0, 0, 0.9, 0.1],
+                                      [0.333333, 0.333333, 0.333333, 0, 0]],
+                                     [[1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0],
+                                      [1, 0, 0, 0, 0],
+                                      [0, 0, 0, 1, 0.0],
+                                      [0, 0, 0, 0.333333, 0.666667]]])
+        weights_cube = self.weights_cube.copy(input_weights)
+        nbhooded_cube = self.weights_cube.copy()
+        plugin = CollapseMaskedNeighbourhoodCoordinate(
+            "projection_x_coordinate", weights_cube)
+        plugin.reweight_weights(nbhooded_cube, weights_cube)
+        self.assertArrayAlmostEqual(expected_weights, weights_cube.data)
+
+
+class Test_process(IrisTest):
```

Comment:
Is there a test for when the weights are None?

Reply:
Not yet. I think we need to support the case where weights are None, but I wanted to think about it a bit more before adding tests.

Reply:
Done